td-layout/src/build_time.rs

```diff
@@ -7,72 +7,76 @@
-|                 CONFIG                 |   (0x40000) 256 KB
+|             LARGE_PAYLOAD              |   (0x0) 0 B
++----------------------------------------+ <- 0x0
+|                 CONFIG                 |   (0x40000) 256 kB
 +----------------------------------------+ <- 0x40000
-|                MAILBOX                 |   (0x1000) 4 KB
+|                MAILBOX                 |   (0x1000) 4 kB
 +----------------------------------------+ <- 0x41000
-|               TEMP_STACK               |   (0x20000) 128 KB
+|               TEMP_STACK               |   (0x20000) 128 kB
 +----------------------------------------+ <- 0x61000
-|               TEMP_HEAP                |   (0x20000) 128 KB
+|               TEMP_HEAP                |   (0x20000) 128 kB
++----------------------------------------+ <- 0x81000
+|                  FREE                  |   (0x0) 0 B
 +----------------------------------------+ <- 0x81000
-|                  FREE                  |   (0x1000) 4 KB
-+----------------------------------------+ <- 0x82000
 |                PAYLOAD                 |   (0xC2D000) 12.18 MB
++----------------------------------------+ <- 0xCAE000
+|                METADATA                |   (0x1000) 4 kB
 +----------------------------------------+ <- 0xCAF000
-|                METADATA                |   (0x1000) 4 KB
-+----------------------------------------+ <- 0xCB0000
-|                  IPL                   |   (0x348000) 3.28 MB
+|                  IPL                   |   (0x349000) 3.29 MB
 +----------------------------------------+ <- 0xFF8000
-|              RESET_VECTOR              |   (0x8000) 32 KB
+|              RESET_VECTOR              |   (0x8000) 32 kB
 +----------------------------------------+ <- 0x1000000
 Image size: 0x1000000 (16 MB)
 */
 
 // Image Layout Configuration
+pub const TD_SHIM_FIRMWARE_SIZE: u32 = 0x1000000;
+
+pub const TD_SHIM_LARGE_PAYLOAD_OFFSET: u32 = 0x0;
+pub const TD_SHIM_LARGE_PAYLOAD_SIZE: u32 = 0x0; // 0 B
 
 pub const TD_SHIM_CONFIG_OFFSET: u32 = 0x0;
-pub const TD_SHIM_CONFIG_SIZE: u32 = 0x40000; // 256 KB
+pub const TD_SHIM_CONFIG_SIZE: u32 = 0x40000; // 256 kB
 
 pub const TD_SHIM_MAILBOX_OFFSET: u32 = 0x40000;
-pub const TD_SHIM_MAILBOX_SIZE: u32 = 0x1000; // 4 KB
+pub const TD_SHIM_MAILBOX_SIZE: u32 = 0x1000; // 4 kB
 
 pub const TD_SHIM_TEMP_STACK_OFFSET: u32 = 0x41000;
-pub const TD_SHIM_TEMP_STACK_SIZE: u32 = 0x20000; // 128 KB
+pub const TD_SHIM_TEMP_STACK_SIZE: u32 = 0x20000; // 128 kB
 
 pub const TD_SHIM_TEMP_HEAP_OFFSET: u32 = 0x61000;
-pub const TD_SHIM_TEMP_HEAP_SIZE: u32 = 0x20000; // 128 KB
+pub const TD_SHIM_TEMP_HEAP_SIZE: u32 = 0x20000; // 128 kB
 
 pub const TD_SHIM_FREE_OFFSET: u32 = 0x81000;
-pub const TD_SHIM_FREE_SIZE: u32 = 0x1000; // 4 KB
+pub const TD_SHIM_FREE_SIZE: u32 = 0x0; // 0 B
 
-pub const TD_SHIM_PAYLOAD_OFFSET: u32 = 0x82000;
+pub const TD_SHIM_PAYLOAD_OFFSET: u32 = 0x81000;
 pub const TD_SHIM_PAYLOAD_SIZE: u32 = 0xC2D000; // 12.18 MB
```

Comment:
You can increase PAYLOAD_SIZE to avoid other changes.

Reply:
Also, please split the "Free Section Removal" change from "Large Payload" change.